msteams-platform/samples/virtual-tables-api.md

```diff
@@ -0,0 +1,229 @@
+  |( )|Precedence grouping |$filter=(m365_name eq ‘Contoso’ and m365_price eq 50.0) or contains(m365_subject,’Team Sync’)|
+
+* **Query Functions**
+
+  |Function |Example |
+  |----|----|
+  |contains|$filter=contains(m365_name,’Contoso’)|
+  |endswith|$filter=endswith(m365_name,’Contoso’)|
+  |startswith|$filter=startswith(m365_name,’Contoso’)|
+
+**Supported Tables**: Each virtual table supports the same filtering functionality as it’s respective Graph resource. The virtual tables, which support filtering are:
+
+* Graph Booking Appointment
+* Graph Drive Item
+* Graph Event
+
+> [!Note]
+> Filtering is not supported on all the attributes of the respective Graph resources. If a user tries to filter on a virtual table with an unsupported attribute, this filter will be ignored. This is the same behaviour as the Dataverse Web API on columns that do not support filtering.
+
+Examples:
+
+* GET [Organization URI]/api/data/v9.2/m365_graphbookingappointments?$filter=m365_bookingbusinessid eq ‘ContosoBank@Contoso.onmicrosoft.com’ and m365_price eq 100.0
+* GET [Organization URI]/api/data/v9.2/m365_graphdriveitems?$filter=m365_collaborationrootid eq ‘00000000-0000-0000-0000-000000000000’ and m365_name eq ‘Meeting Notes.docx’
+* GET [Organization URI]/api/data/v9.2/m365_graphevents?$filter=m365_groupid eq ‘00000000-0000-0000-0000-000000000000’ and m365_subject eq ‘Monthly Sync’
+
+## Virtual table pagination
+
+Pagination is a useful resource for fetching a large set of records. Virtual Table pagination can be achieved in three different ways.
+
+You can specify the page size by using the `odata.maxpagesize` preference value in the request header. If the result set spans multiple pages, the response will include the `@odata.nextLink` property. Sample request and response are as following:
+
+# [Request](#tab/request)
+
+```http
+  GET [Organization URI]/api/data/v9.2/m365_driveitems 
+  Accept: application/json 
+  Prefer: odata.maxpagesize=2 
+```
+
+# [Response](#tab/response)
+
+```json
+{ 
+
+  "@odata.context": "[Organization URI]/api/data/v9.0/$metadata#m365_graphdriveitems", 
+  "value": [ 
+   { 
+      "@odata.etag": "W/\"{FA93AF7C-1F45-4714-85A5-BB95EB86E1E5}\"", 
+      "m365_name": "Review.doc", 
+      "m365_graphdriveitemid": "f50aae23-6644-3d35-66d7-e3c5a979dad3", 
+      …
+      },
+      {
+      "@odata.etag": "W/\"{3938D549-1AEF-46A5-BF3C-38472AD934C2}\"", 
+      "m365_name": "Review.doc", 
+      "m365_graphdriveitemid": "3d59a7e2-ec83-d0b3-270e-8ad676622027", 
+      … 
+      } 
+      ],
+      "@odata.nextLink": "[Organization URI]/api/data/v9.0/m365_graphdriveitems &$skiptoken=%3Ccookie%20pagenumber=%222%22%20pagingcookie=%22UGFnZWQ9VFJVRSZwX1NvcnRCZWhhdmlvcj0xJnBfRmlsZUxlYWZSZWY9dGVzdCZwX0lEPTI5%22%20istracking=%22False%22%20/%3E" 
+} 
+```
+
+---
+
+Currently the following Virtual Tables support the `odata.maxpagesize` preference:
+
+* Graph Booking Appointment
+* Graph Calendar Event
+* Graph Chat
+* Graph Chat Message
```

Comment:
Graph Chat and Graph Chat Message tables won't be supported, so they can be removed.